filters/ELMFilter.cpp

```diff
@@ -122,16 +116,61 @@ void ELMFilter::filter(PointView& view)
 
                 // Otherwise this point is classified as noise, and we proceed
                 // to the next lowest value.
-                view.setField(Dimension::Id::Classification, it->second,
+                view.setField(m_Classdim, it->second,
                               m_class);
                 ++num;
             }
         }
     }
 
+    return num;
+
+}
+
+
+void ELMFilter::filter(PointView& view)
+{
+    BOX2D bounds;
+    calculateBounds(view, bounds);
+
+    size_t cols =
```

Comment:
This is all pretty expensive as it takes three passes through the points.  One could certainly eliminate the bounds calculation -- you don't need a full grid, you just need cells.